### PR TITLE
GPTBridge-21: Tool outputs

### PR DESCRIPTION
**Pull request by froggomad**
# Story
[Submit tool call outputs and cancel run](https://infinitiqtech.atlassian.net/browse/GPTBRIDGE-21)

# Description
This is a big one, bear with me!

1. Implements `submitToolOutputs`
- this is used when the assistant should know about the results of function calls it generated in order to continue processing a run.

2. Implements `cancelRun`
- this is used to end a run for any reason, but especially when the assistant has run a function and you don't want the assistant to be aware of the result.

# Pre-Requisites
To test, you must have an OpenAI API Key and assistant ID. The API key can be created in the assistant's API and the assistant's key can be found in the API as well

# Checklist
**Required**
- [x] Implemented a unit test
- [ ] Modified a unit test
- [ ] Didn't require unit testing
  - Explain briefly

**Optional**
- [x] Added a 3rd-party dependency
  - Docc for Swift Documentation
- [x] Modified the public API
  - added cancel run
  - added submitToolOutputs

## Summary by CodeRabbit

- **New Features**
	- Introduced a new dashboard application, "My GPT Dashboard," for managing and interacting with assistants using the GPTBridge API.
	- Added functionality to list and interact with assistants in a user-friendly interface.
	- Enhanced the pull request template with additional checklist sub-items for better clarity on changes.
	- Expanded the GPT-Bridge framework with new capabilities such as listing assistants and submitting tool outputs.
	- Improved interaction with the OpenAI assistant's API through the GPT-Bridge framework, including detailed guidelines and usage steps.
- **Bug Fixes**
	- Improved error handling across the application and framework, adding new error cases for better clarity and troubleshooting.
- **Documentation**
	- Updated README.md with comprehensive installation instructions, setup guidelines, and usage steps for the GPT-Bridge framework.
- **Refactor**
	- Several internal improvements and code refactoring for better maintainability and performance, including changes in handling secrets and API keys.
- **Tests**
	- Added new test cases for function argument decoding, JSON serialization/deserialization, listing assistants, and submitting tool calls.

### CODE REVIEW

**Comment by froggomad** [comment]
@coderabbitai review

File: Sources/GPTBridge/Models/RunHandlers/Interface/RunHandler.swift [hunk 1]
```diff
@@ -13,3 +13,28 @@ protocol RunHandler {
...
+public struct FunctionRunStepResult: RunStepResult {
+    public let toolCallId: String
+    public let functions: [AssistantFunction]?
+    public let message: String? = nil
+
+    init(toolCallId: String, functions: [AssistantFunction]) {
+        self.toolCallId = toolCallId
+        self.functions = functions
+    }
+}
```
[the remark this answers is not in the document]
Comment (froggomad) [reply Sources/GPTBridge/Models/RunHandlers/Interface/RunHandler.swift:31]: I purposefully left it internal so the user doesn't try to instantiate one. This is the result of a public method. Your thoughts?

File: Sources/GPTBridge/Models/RunHandlers/Interface/RunHandler.swift [hunk 2]
```diff
@@ -13,3 +13,28 @@ protocol RunHandler {
...
+
+struct MessageRunStepResult: RunStepResult {
+    var functions: [AssistantFunction]? = nil
+    var message: String?
+
+    init(message: String) {
+        self.message = message
+    }
+}
```
[the remark this answers is not in the document]
Comment (froggomad) [reply Sources/GPTBridge/Models/RunHandlers/Interface/RunHandler.swift:40]: this is an internal struct

File: My GPT Dashboard/My GPT Dashboard.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+		AA770D9E2BC113BD00D7FAE6 /* My_GPT_DashboardApp.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = My_GPT_DashboardApp.swift; sourceTree = "<group>"; };
+		AA770DA02BC113BD00D7FAE6 /* AssistantListView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = AssistantListView.swift; sourceTree = "<group>"; };
+		AA770DA22BC113BE00D7FAE6 /* Assets.xcassets */ = {isa = PBXFileReference; lastKnownFileType = folder.assetcatalog; path = Assets.xcassets; sourceTree = "<group>"; };
+		AA770DA42BC113BE00D7FAE6 /* My_GPT_Dashboard.entitlements */ = {isa = PBXFileReference; lastKnownFileType = text.plist.entitlements; path = My_GPT_Dashboard.entitlements; sourceTree = "<group>"; };
+		AA770DA62BC113BE00D7FAE6 /* Preview Assets.xcassets */ = {isa = PBXFileReference; lastKnownFileType = folder.assetcatalog; path = "Preview Assets.xcassets"; sourceTree = "<group>"; };
+		AA770DAD2BC113F600D7FAE6 /* GPTBridge */ = {isa = PBXFileReference; lastKnownFileType = wrapper; name = GPTBridge; path = /Users/kennethdubroff/Development/Xcode_Development/Projects/GPTBridge; sourceTree = "<absolute>"; };
```
[the remark this answers is not in the document]
Comment (froggomad) [reply My GPT Dashboard/My GPT Dashboard.xcodeproj/project.pbxproj:24]: pbxproj is generated by XCode

**Comment by brookshanes** [comment]
> OpenAI

wowza

**Pull request merged**